src/main/java/org/prebid/server/bidder/rubicon/RubiconBidder.java

```diff
@@ -1266,27 +1273,98 @@ private void mergeFirstPartyDataFromUser(ExtUser userExt, ObjectNode result) {
         }
     }
 
-    private static void enrichWithIabAttribute(ObjectNode target, List<Data> data, Set<Integer> segtaxValues) {
-        final List<String> iabValue = CollectionUtils.emptyIfNull(data).stream()
-                .filter(Objects::nonNull)
-                .filter(dataRecord -> containsSegtaxValue(dataRecord.getExt(), segtaxValues))
-                .map(Data::getSegment)
+    private static void enrichWithIabAndSegtaxAttribute(ObjectNode target, List<Data> data, Set<Integer> segtaxValues) {
+        final Map<Integer, List<Segment>> segments = CollectionUtils.emptyIfNull(data).stream()
                 .filter(Objects::nonNull)
-                .flatMap(segments -> segments.stream()
-                        .map(Segment::getId))
+                .map(RubiconBidder::getValidSegments)
                 .filter(Objects::nonNull)
-                .toList();
+                .collect(Collectors.toMap(
+                        Map.Entry::getKey,
+                        Map.Entry::getValue,
+                        (value1, value2) -> Stream.concat(value1.stream(), value2.stream())
+                                .collect(Collectors.toCollection(ArrayList::new))));
```

Comment:
If we are moving away from lists:
```java
                        (first, second) -> {
                            first.addAll(second);
                            return first;
                        }));
```

Also, please, place this methods in call order:
```
    enrichWithIabAndSegtaxAttribute

    getValidSegments

    getValidOnlySegments
 
    pickRelevantSegments

    groupBySegtaxValues  

    resolveSegmentName
```